cogs/tibia.py

```diff
@@ -1932,6 +1932,32 @@ def __unload(self):
         print("cogs.tibia: Cancelling pending tasks...")
         self.news_announcements_task.cancel()
 
+    @checks.is_mod()
+    @commands.command(name="respawnarea", aliases=['resparea'])
+    async def respawn_area(self, ctx: NabCtx):
+        """Announces the respawn area if there is a channel set. Area is chosen by mods."""
+        guild = ctx.guild
+        channel_id = await get_server_property(ctx.pool, guild.id, "channel_respawn_area")
+        if channel_id == 0:
+            await ctx.send("You need to configure a channel for this first. Use `/settings respawnareachannel` for "
+                           + "more information.")
+            return
+        if ctx.is_private or ctx.channel.name != config.respawn_area_channel_name:
```

Comment:
I'm confused, why do you need a channel just to use the command to set an area, and a separate to see the new respawn areas?

Reply:
That was not the intention and, as far as I could test, that is also not happening.
Maybe the variable names are confusing? Why do you think this is the case?

Considering the user set the channel to `#respawn-area` with `/settings`, my intention is to not allow user to call this command anywhere but `#respawn-area`. This is to avoid general users seeing random commands with no answer in other channels, as the bot erases his answers from where it was used, and announces the area in `#respawn-area`, leaving just the command call behind. It would look bad and I do not see any benefit of allowing mods to use the command in any other channel.

Reply:
Because at the beginning of this command, you check the value of `channel_respawn_area`, but then, in this condition, you're checking if the channel's name matches the name set in config.yml

So, if you used `/settings` to set the channel to something not named the same as `config.respawn_area_channel_name`, then you can't get past this condition.

Reply:
I will remove it from the `config.yml` file then, and let the user set it through `/settings` as the first message already tells him to.

Reply:
Removed `configs`, comparing with channel ID now.